Lombiq.Tests.UI/Extensions/NavigationUITestContextExtensions.cs

```diff
@@ -204,26 +213,51 @@ public static Task<Uri> GoToSetupPageAndSetupOrchardCoreAsync(this UITestContext
 
     public static async Task<Uri> GoToSetupPageAndSetupOrchardCoreAsync(
         this UITestContext context,
-        OrchardCoreSetupParameters parameters = null)
+        OrchardCoreSetupParameters parameters = null,
+        bool shouldBeSuccess = true)
     {
-        var setupPage = await context.GoToSetupPageAsync(parameters?.RunSetupOnCurrentPage == false);
-        setupPage = await setupPage.SetupOrchardCoreAsync(context, parameters);
+        parameters ??= new(context);
+
+        if (!parameters.RunSetupOnCurrentPage) await context.GoToAbsoluteUrlAsync(context.TestStartUri);
+        await parameters.SetupOrchardCoreAsync(context);
+        context.CheckExistence(OrchardCoreSetupParameters.FinishSetupSelector, !shouldBeSuccess);
 
-        return setupPage.PageUri.Value;
+        return new(context.Driver.Url);
     }
 
+    [Obsolete("Methods using Page<> classes will be removed in the next version.")]
     public static Task<OrchardCoreRegistrationPage> GoToRegistrationPageAsync(this UITestContext context) =>
         context.GoToPageAsync<OrchardCoreRegistrationPage>();
 
+    [Obsolete($"Use {nameof(GoToAdminAsync)} method instead.")]
```

Comment:
Add new versions for the other methods that have "dashboard" in the name too and obsolete the other ones.

But. There are a lot of places where we refer to the admin area as dashboard, starting with the OC UI itself, of course. But therre's also `OrchardCoreDashboardUITestContextExtensions` and a lot of other UI testing-related stuff with the same terminology. So, it would better to keep that here too.

Reply:
I don't know. My best suggestion is break this by removing the Page return type, and adding a temporary, obsolete `GoToDashboardPageAsync` for those using this.